### PR TITLE
Added reset button for file fields

### LINKED ISSUE #4762: Add 'Remove file' button if user selected file for product with customizable option on PDP
**Description**:
Add 'Remove file' button if user selected file for product with customizable option on PDP

**Expected behavior**:
Remove this logic - validation happens when clicking on file input
Add 'Remove file' button if file selected similarly to in screenshot 
Screenshot 2022-05-09 at 13 08 46

**To Reproduce**:
1. Open product with required file option - https://master.d25kd40decjpue.amplifyapp.com/default/testing-simple.html
2. Click on file input

**Actual result**:
File input is validated 
Screenshot 2022-05-09 at 13 12 40

### PR DESCRIPTION
**Pull request by AzizKHAN030**
**Related issue(s):**
* Fixes #4762 

**Problem:**
* The validation and field reset was on click to the fields which looked not very good

**In this PR:**
* Added a button for the fields which resets the field and validates after

### CODE REVIEW

**Comment by Elena-Evsa** [comment]
1. Styling of remove button is not the same as in expected result
2. 'Remove file' is blue on IOS 15 
image
3. When user is clicking on 'Remove file' on IOS then the same behavior as user clicking on file field: fields loads the options with file sources https://vimeo.com/708617716/0813e40eb6
4. Big indent between error message under field and text with compatible file extensions
image

File: packages/scandipwa/src/component/Field/Field.style.scss [hunk 1]
```diff
@@ -268,6 +268,15 @@
                             word-break: break-word;
                             text-align: center;
                             border-color: var(--primary-success-color);
+
+                            & + button {
+                                border: 2px solid var(--primary-success-color);
+
+                                &:hover {
+                                    background-color: var(--primary-success-color);
+                                    color: #{$white};
```
Comment (zans-laksa) [inline packages/scandipwa/src/component/Field/Field.style.scss:277]: Why the $white variable doesn't have the # character so you should put it separately?

File: packages/scandipwa/src/component/FieldFile/FieldFile.container.js [hunk 2]
```diff
@@ -119,9 +119,9 @@ export class FieldFileContainer extends PureComponent {
             setRef,
             events: {
                 ...events,
-                onChange: this.onChange.bind(this),
-                onClick: resetFieldValue.bind(this, this)
+                onChange: this.onChange.bind(this)
             },
+            resetFieldValue: resetFieldValue.bind(this, this),
```
Comment (zans-laksa) [inline packages/scandipwa/src/component/FieldFile/FieldFile.container.js:124]: Please check the reason why `this` is passed twice. I believe this is a bug.

**Review by zans-laksa — CHANGES_REQUESTED** [review]

File: packages/scandipwa/src/component/Field/Field.style.scss (hunk 1 above)
Comment (zans-laksa) [reply packages/scandipwa/src/component/Field/Field.style.scss:277]: Please put this color instead: `var(--color-white)`

File: packages/scandipwa/src/component/Field/Field.style.scss [hunk 3]
```diff
@@ -278,9 +287,26 @@
                             word-break: break-word;
                             text-align: center;
                             border-color: var(--primary-error-color);
+
+                            & + button {
+                                border: 2px solid var(--primary-error-color);
+
+                                &:hover {
+                                    background-color: var(--primary-error-color);
+                                    color: #{$white};
```
Comment (zans-laksa) [inline packages/scandipwa/src/component/Field/Field.style.scss:296]: var(--color-white)

File: packages/scandipwa/src/component/Field/Field.style.scss (hunk 1 above)
Comment (zans-laksa) [reply packages/scandipwa/src/component/Field/Field.style.scss:277]: Please put this color: `var(--color-white)`

**Review by zans-laksa — APPROVED** [review]

File: packages/scandipwa/src/component/FieldFile/FieldFile.container.js (hunk 2 above)
Comment (zans-laksa) [reply packages/scandipwa/src/component/FieldFile/FieldFile.container.js:124]: @AzizKHAN030 Did you check this? What was the reason to pass this twice?

Comment (AzizKHAN030) [reply packages/scandipwa/src/component/FieldFile/FieldFile.container.js]: I passed the field context as a parameter in order to set the state of the field from the parent component. Now I refactored it and I'm passing an object that has a method which calls setState.

**Comment by carinadues** [comment]
1. Failed. The styling of the 'Remove file' button should be like the expected result:
image

2. Passed
3. Passed
4. Passed

**Review by zans-laksa — APPROVED** [review]

**Pull request merged**